tests/testthat/test-betydb.R

```diff
@@ -127,6 +129,44 @@ test_that("betydb_query works", {
   expect_true(all(np_grass$id %in% np$id))
 })
 
+test_that("paging works with betydb query and search functions",{
+  skip_on_cran()
+  check_betydb()
+
+  options(
+    betydb_url = "https://www.betydb.org/",
+    betydb_api_version = "beta",
+    betydb_key = "eI6TMmBl3IAb7v4ToWYzR0nZYY07shLiCikvT6Lv",
+    warn=-1 ## suppress warnings that we did not get all data
+  )
+
+  # check paging much faster than default hardcoded limit of 5000
+  per_call_limit <<- 10
+
+  # return 200 records by default
+  limit_default <- betydb_query(table = "traits")
+  expect_equal(nrow(limit_default), 200)
+
+  # check that paging returns correct # below and above default
+  limit3 <- betydb_query(table = 'traits', limit = 3)
+  expect_equal(nrow(limit3), 3)
+  expect_equal(nrow(limit3), attributes(limit3)$metadata$count)
+
+  limit30 <- betydb_query(table = "traits", limit = 30)
+  expect_equal(nrow(limit30), 30)
+  expect_equal(nrow(limit30), attributes(limit30)$metadata$count)
+
+  limit401 <- betydb_query(table = 'traits', limit = 401)
+  expect_equal(nrow(limit401), 401)
+  expect_equal(nrow(limit401), attributes(limit401)$metadata$count)
+
+})
+
+options(
+  betydb_url = "https://www.betydb.org/",
+  betydb_api_version = "v0",
+  warnings = 0)
+
```

Comment:
Ah, OK. This works, but `on.exit(reset_opts(opts))` is still arguably safer: It both *resets* all previous options, and *unsets* any newly set options that didn't exist before this test.